GameDesign/Assets/Scripts/MonoBehaviours/Movements/TalkingWayPoint.cs

```diff
@@ -0,0 +1,38 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class TalkingWayPoint : Waypoint
+{
+    [Range(0f, 100f)]
+    [SerializeField] private float talkingTime = 1.2f;
+    [SerializeField] Transform lookAtTarget;
+    public override void waypointReached(NPCMovementManager npcMover)
+    {
+        StartCoroutine(Talking(npcMover));
+    }
+
+    IEnumerator Talking(NPCMovementManager npcMover)
+    {
+
+
+        Animator animator = npcMover.GetComponent<Animator>();
+        if (animator != null)
+        {
+            //TODO make smooth transition instead of sudden movement
```

Comment:
Is it something we can do easily or better to focus on other stuff in the meantime?

Reply:
easily a prefab is already done takes a couple mins to add all of it

Reply:
so let's do it before merging

Reply:
Ok, but after #127 since its adding waypoints to the scenes. And it makes sense to do it to the set levels

Reply:
Ah, so let's merge it and create an issue to add the workers